pallets/offchain-demo/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg(test)]
+mod tests;
+
+use support::{
+	debug,
+	dispatch::DispatchResult, decl_module, decl_storage, decl_event, decl_error,
+	traits::Get,
+	weights::SimpleDispatchInfo,
+};
+
+use core::convert::{TryInto};
+
+use system::{self as system, ensure_signed, ensure_none, offchain};
+// use serde_json as json;
+use sp_core::crypto::KeyTypeId;
+use sp_runtime::{
+	transaction_validity::{InvalidTransaction, ValidTransaction, TransactionValidity},
+};
+use sp_std::prelude::*;
+
+/// Defines application identifier for crypto keys of this module.
+///
+/// Every module that deals with signatures needs to declare its unique identifier for
+/// its crypto keys.
+/// When offchain worker is signing transactions it's going to request keys of type
+/// `KeyTypeId` from the keystore and use the ones it finds to sign the transaction.
+/// The keys can be inserted manually via RPC (see `author_insertKey`).
+pub const KEY_TYPE: KeyTypeId = KeyTypeId(*b"demo");
+pub const NUM_VEC_LEN: usize = 10;
+
+/// Based on the above `KeyTypeId` we need to generate a pallet-specific crypto type wrappers.
+/// We can use from supported crypto kinds (`sr25519`, `ed25519` and `ecdsa`) and augment
+/// the types with this pallet-specific identifier.
+pub mod crypto {
+	use crate::KEY_TYPE;
+	use sp_runtime::app_crypto::{app_crypto, sr25519};
+	app_crypto!(sr25519, KEY_TYPE);
+}
+
+/// This is the pallet's configuration trait
+pub trait Trait: system::Trait {
+	/// The overarching dispatch call type.
+	type Call: From<Call<Self>>;
+	/// The overarching event type.
+	type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+	/// The type to sign and submit transactions.
+	type SubmitSignedTransaction: offchain::SubmitSignedTransaction<Self, <Self as Trait>::Call>;
+	/// The type to submit unsigned transactions.
+	type SubmitUnsignedTransaction: offchain::SubmitUnsignedTransaction<Self, <Self as Trait>::Call>;
+	type GracePeriod: Get<Self::BlockNumber>;
+}
+
+// Custom data type
+#[derive(Debug)]
+enum TransactionType {
+	Signed,
+	Unsigned,
+	None,
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as Example {
+		/// A vector of recently submitted prices. Should be bounded
+		Numbers get(fn numbers): Vec<u64>;
+
+		/// Defines the block when next unsigned transaction will be accepted.
+		NextTx get(fn next_tx): T::BlockNumber;
+		AddSeq get(fn add_seq): u32;
+	}
+}
+
+decl_event!(
+	/// Events generated by the module.
+	pub enum Event<T> where AccountId = <T as system::Trait>::AccountId {
+		/// Event generated when new price is accepted to contribute to the average.
+		NewNumber(Option<AccountId>, u64),
+	}
+);
+
+decl_error! {
+	pub enum Error for Module<T: Trait> {
+		SendSignedError,
+		SendUnsignedError
```

Comment:
Should have documentation on these errors.
```suggestion
		SendUnsignedError,
```